uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -607,25 +594,39 @@ define([
             return self.directiveOptions.forStaff && loadManagees.call(self);
           })
           .then(function () {
-            if (self.directiveOptions.forStaff && !self.contact) {
-              //throw will break out of chain
-              throw "You have to select a staff to apply absence on their behalf.";
-            }
-
             return loadAbsencePeriods.call(self);
           })
           .then(function () {
             initAbsencePeriod.call(self);
-            self._setMinMaxDate.call(self);
+            self._setMinMaxDate();
 
             return $q.all([
-              self._loadAbsenceTypes.call(self),
               loadCommentsAndContactNames.call(self),
-              self.request.loadAttachments(),
-              self._loadCalendar.call(self)
+              self.request.loadAttachments()
             ]);
           })
           .then(function () {
+            if (self.directiveOptions.forStaff && !self.request.contact_id) {
+              return;
+            }
+
+            return self.initAfterContactSelection();
+          });
+      };
+
+      /**
+       * Initializes after contact is selected either directly or by manager
+       *
+       * @return {Promise}
+       */
+      this.initAfterContactSelection = function () {
```

Comment:
there doesn't seem to be any reason for this method to be public, it's just an internal method